app/models/concerns/host_params.rb

```diff
@@ -8,59 +8,65 @@ module HostParams
     include ParameterValidators
     attr_reader :cached_host_params
 
-    def params
-      host_params.update(lookup_keys_params)
-    end
+    register_info_provider :host_params_info
+  end
 
-    def clear_host_parameters_cache!
-      @cached_host_params = nil
-    end
+  def host_params_info
+    { 'parameters' => params }
+  end
 
-    def host_inherited_params(include_source = false)
-      hp = {}
-      params = host_inherited_params_objects
-      params.each do |param|
-        source = param.associated_type
-        options = {:value => param.value,
-                   :source => source,
-                   :safe_value => param.safe_value }
-        if source != 'global'
-          options.merge!(:source_name => param.associated_label)
-        end
-        hp.update(Hash[param.name => include_source ? options : param.value])
-      end
-      hp
-    end
+  def params
+    host_params
+  end
 
-    def host_params
-      return cached_host_params unless cached_host_params.blank?
-      hp = host_inherited_params
-      # and now read host parameters, override if required
-      host_parameters.authorized(:view_params).each { |p| hp[p.name] = p.value }
-      @cached_host_params = hp
-    end
+  def clear_host_parameters_cache!
+    @cached_host_params = nil
+  end
 
-    def host_inherited_params_objects
-      params = CommonParameter.all
-      if SETTINGS[:organizations_enabled] && organization
-        params += extract_params_from_object_ancestors(organization)
+  def host_inherited_params(include_source = false)
+    hp = {}
+    params = host_inherited_params_objects
+    params.each do |param|
+      source = param.associated_type
+      options = {:value => param.value,
+                 :source => source,
+                 :safe_value => param.safe_value }
+      if source != 'global'
+        options.merge!(:source_name => param.associated_label)
       end
+      hp.update(Hash[param.name => include_source ? options : param.value])
+    end
+    hp
+  end
 
-      if SETTINGS[:locations_enabled] && location
-        params += extract_params_from_object_ancestors(location)
-      end
+  def host_params
+    return cached_host_params unless cached_host_params.blank?
+    hp = host_inherited_params
+    # and now read host parameters, override if required
+    host_parameters.authorized(:view_params).each { |p| hp[p.name] = p.value }
+    @cached_host_params = hp
```

Comment:
It's a bit risky, since it would be shared between instances. For example look at the use case of changing the hostgroup: first you will get an existing instance of the host (to show it in the UI) the cache is created (with values for the old hostgroup); now the user changes the hostgroup and the AJAX call is performed - a new host instance is created, but if the cached values will be used it will be wrong.